src/DepositHandler.sol

```diff
@@ -22,16 +24,23 @@ import {IDepositHandlerErrors} from "./interfaces/ICustomErrors.sol";
  */
 contract DepositHandler is Pausable, AccessControl, IDepositHandlerErrors {
     bytes32 public constant MANAGER = keccak256("MANAGER");
+    uint256 public constant DEPOSIT_STAGE_DURATION = 2 days;
```

Comment:
~This should certainly be defined during the creation (argument of the constructor) ?~

Reply:
After reviewing the rest I think we should simply delete it (since it make sense with your initial thoughts of on chain whitelisting but not with an off-chain approval). But I understand where this come from ; )

Reply:
This doesn't relates to whitelisting, this is a restriction for accepted users to do a deposit in time, to avoid situation of depositing tokens in the middle of the bootcamp for example. 

More deep explanation is in the next your comment)